mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageManager.java

```diff
@@ -427,12 +427,9 @@ public ComposedMessageId appendMessage(InputStream msgIn, Date internalDate, fin
                 public ComposedMessageId execute() throws MailboxException {
                     MessageMetaData data = appendMessageToStore(message, attachments, mailboxSession);
 
-                    SortedMap<MessageUid, MessageMetaData> uids = new TreeMap<MessageUid, MessageMetaData>();
-                    MessageUid messageUid = data.getUid();
-                    MailboxId mailboxId = getMailboxEntity().getMailboxId();
-                    uids.put(messageUid, data);
-                    dispatcher.added(mailboxSession, uids, getMailboxEntity());
-                    return new ComposedMessageId(mailboxId, data.getMessageId(), messageUid);
+                    Mailbox mailbox = getMailboxEntity();
+                    dispatcher.added(mailboxSession, message, data, mailbox);
+                    return new ComposedMessageId(mailbox.getMailboxId(), data.getMessageId(), data.getUid());
```

Comment:
We can't, the MessageMapper methods for copy & move are returning MessageMetaData.
So we have to rewrite the mapper methods ...